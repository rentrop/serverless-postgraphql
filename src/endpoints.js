
var endpoint = 'http://localhost:5000/graphql';

if(process.env.USE_AWS_ENDPOINT) {
  endpoint = process.env.AWS_ENDPOINT;
};

module.exports = {
    endpoint: endpoint
};
