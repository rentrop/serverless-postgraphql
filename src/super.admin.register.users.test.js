import HttpTransport from 'lokka-transport-http';
import Lokka from 'lokka';
import uuidv4 from 'uuid';
import { endpoint } from './endpoints';

const anonLokka = new Lokka({transport: new HttpTransport(endpoint)});
const superAdminEmail = 'superadmin@flo.ods';
const superAdminPassword = 'texasfloods';
const newUserPassword = 'texasfloods';

async function getToken(email, password) {
  const response = await anonLokka.send(`
    mutation($email:String!, $password:String!) {
      authenticate(input: {email: $email, password: $password}) {
        jwtToken
      }
    }
  `,
  {
    email: email,
    password: password
  });

  return response.authenticate.jwtToken;
}

describe('When registering a super admin', () => {
  var newUserEmail = uuidv4() + '@flo.ods';
  var newUserId;

  describe('As a super admin', async () => {
    var lokka;

    beforeAll(async (done) => {
      getToken(superAdminEmail, superAdminPassword).then((token) => {
        const headers = {
          'Authorization': 'Bearer '+ token
        };
        lokka = new Lokka({transport: new HttpTransport(endpoint, {headers})});
        done();
      });
    });

    it('should register a new super admin', async () => {
      const response = await lokka.send(`
        mutation($email:String!) {
          registerUser(input: {
            firstName: "New",
            lastName: "User",
            jobTitle: "Super Admin",
            communityId: 1,
            phoneNumber: "555-6666",
            email: $email,
            password:"texasfloods",
            role:"floods_super_admin"
          }) {
            user {
              id
            }
          }
        }
      `,
      {
        email: newUserEmail
      });

      newUserId = response.registerUser.user.id;
      expect(response).not.toBeNull();
    });

  });

  it('the new user should show up in the DB', async () => {
    const response = await anonLokka.send(`
      query ($id: Int!) {
        userById(id: $id) {
          firstName
          lastName
          jobTitle
        }
      }
    `,
    {
      id: newUserId
    });

    expect(response).toMatchSnapshot();
  });

  describe('As the new super admin', async () => {
    var lokka;

    beforeEach(async (done) => {
      getToken(newUserEmail, newUserPassword).then((token) => {
        const headers = {
          'Authorization': 'Bearer '+ token
        };
        lokka = new Lokka({transport: new HttpTransport(endpoint, {headers})});
        done();
      });
    });

    it('should get the correct current user', async () => {
      const response = await lokka.send(`
        {
          currentUser {
            firstName
            lastName
            jobTitle
          }
        }
      `);

      expect(response).toMatchSnapshot();
    });
  });
});

describe('When registering a community admin', () => {
  var newUserEmail = uuidv4() + '@flo.ods';
  var newUserId;

  describe('As a super admin', async () => {
    var lokka;

    beforeAll(async (done) => {
      getToken(superAdminEmail, superAdminPassword).then((token) => {
        const headers = {
          'Authorization': 'Bearer '+ token
        };
        lokka = new Lokka({transport: new HttpTransport(endpoint, {headers})});
        done();
      });
    });

    it('should register a new community admin', async () => {
      const response = await lokka.send(`
        mutation($email:String!) {
          registerUser(input: {
            firstName: "New",
            lastName: "User",
            jobTitle: "Community Admin",
            communityId: 1,
            phoneNumber: "555-6666",
            email: $email,
            password:"texasfloods",
            role:"floods_community_admin"
          }) {
            user {
              id
            }
          }
        }
      `,
      {
        email: newUserEmail
      });

      newUserId = response.registerUser.user.id;
      expect(response).not.toBeNull();
    });

  });

  it('the new user should show up in the DB', async () => {
    const response = await anonLokka.send(`
      query ($id: Int!) {
        userById(id: $id) {
          firstName
          lastName
          jobTitle
        }
      }
    `,
    {
      id: newUserId
    });

    expect(response).toMatchSnapshot();
  });

  describe('As the new community admin', async () => {
    var lokka;

    beforeEach(async (done) => {
      getToken(newUserEmail, newUserPassword).then((token) => {
        const headers = {
          'Authorization': 'Bearer '+ token
        };
        lokka = new Lokka({transport: new HttpTransport(endpoint, {headers})});
        done();
      });
    });

    it('should get the correct current user', async () => {
      const response = await lokka.send(`
        {
          currentUser {
            firstName
            lastName
            jobTitle
          }
        }
      `);

      expect(response).toMatchSnapshot();
    });
  });
});

describe('When registering a community editor', () => {
  var newUserEmail = uuidv4() + '@flo.ods';
  var newUserId;

  describe('As a super admin', async () => {
    var lokka;

    beforeAll(async (done) => {
      getToken(superAdminEmail, superAdminPassword).then((token) => {
        const headers = {
          'Authorization': 'Bearer '+ token
        };
        lokka = new Lokka({transport: new HttpTransport(endpoint, {headers})});
        done();
      });
    });

    it('should register a new community editor', async () => {
      const response = await lokka.send(`
        mutation($email:String!) {
          registerUser(input: {
            firstName: "New",
            lastName: "User",
            jobTitle: "Community Editor",
            communityId: 1,
            phoneNumber: "555-6666",
            email: $email,
            password:"texasfloods",
            role:"floods_community_editor"
          }) {
            user {
              id
            }
          }
        }
      `,
      {
        email: newUserEmail
      });

      newUserId = response.registerUser.user.id;
      expect(response).not.toBeNull();
    });

  });

  it('the new user should show up in the DB', async () => {
    const response = await anonLokka.send(`
      query ($id: Int!) {
        userById(id: $id) {
          firstName
          lastName
          jobTitle
        }
      }
    `,
    {
      id: newUserId
    });

    expect(response).toMatchSnapshot();
  });

  describe('As the new community editor', async () => {
    var lokka;

    beforeEach(async (done) => {
      getToken(newUserEmail, newUserPassword).then((token) => {
        const headers = {
          'Authorization': 'Bearer '+ token
        };
        lokka = new Lokka({transport: new HttpTransport(endpoint, {headers})});
        done();
      });
    });

    it('should get the correct current user', async () => {
      const response = await lokka.send(`
        {
          currentUser {
            firstName
            lastName
            jobTitle
          }
        }
      `);

      expect(response).toMatchSnapshot();
    });
  });
});
