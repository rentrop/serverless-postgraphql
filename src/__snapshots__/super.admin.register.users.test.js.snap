// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When registering a community admin As the new community admin should get the correct current user 1`] = `
Object {
  "currentUser": Object {
    "firstName": "New",
    "jobTitle": "Community Admin",
    "lastName": "User",
  },
}
`;

exports[`When registering a community admin the new user should show up in the DB 1`] = `
Object {
  "userById": Object {
    "firstName": "New",
    "jobTitle": "Community Admin",
    "lastName": "User",
  },
}
`;

exports[`When registering a community editor As the new community editor should get the correct current user 1`] = `
Object {
  "currentUser": Object {
    "firstName": "New",
    "jobTitle": "Community Editor",
    "lastName": "User",
  },
}
`;

exports[`When registering a community editor the new user should show up in the DB 1`] = `
Object {
  "userById": Object {
    "firstName": "New",
    "jobTitle": "Community Editor",
    "lastName": "User",
  },
}
`;

exports[`When registering a super admin As the new super admin should get the correct current user 1`] = `
Object {
  "currentUser": Object {
    "firstName": "New",
    "jobTitle": "Super Admin",
    "lastName": "User",
  },
}
`;

exports[`When registering a super admin the new user should show up in the DB 1`] = `
Object {
  "userById": Object {
    "firstName": "New",
    "jobTitle": "Super Admin",
    "lastName": "User",
  },
}
`;
